src/components/panel/panel.js

```diff
@@ -1160,3 +1234,128 @@ MdPanelPosition.prototype.right = function(opt_right) {
+ * });
+ *
+ * @final @constructor
+ */
+function MdPanelAnimation() {
+  /** @private {!Object} */
+  this._openFrom = {};
+
+  /** @private {!Object} */
+  this._closeTo = {};
+}
+
+
+/**
+ * Gets the boundingClientRect for the opening animation.
+ * @returns {!Object}
+ */
+MdPanelAnimation.prototype.getOpenFrom = function() {
+  return this._openFrom;
+};
+
+
+/**
+ * Gets the boundingClientRect for the closing animation.
+ * @returns {!Object}
+ */
+MdPanelAnimation.prototype.getCloseTo = function() {
+  return this._closeTo;
+};
+
+
+/**
+ * Specifies where to start the open animation. `openFrom` accepts a
+ * click event object, query selector, DOM element, or a Rect object that
+ * is used to determine the bounds. When passed a click event, the location
+ * of the click will be used as the position to start the animation.
+ *
+ * @param {string|!Element|!jQuery.Event|
+ *     {top: string, left: string, height: string, width: string}} openFrom
+ * @returns {MdPanelAnimation}
+ */
+MdPanelAnimation.prototype.openFrom = function(openFrom) {
+  // Check if 'openFrom' is a jQuery.Event.
+  openFrom = openFrom.target ? openFrom.target : openFrom;
+
+  this._openFrom = this._getBoundingClientRect(this._getElement(openFrom));
+
+  if (angular.isUndefined(this._closeTo)) {
+    this._closeTo = angular.copy(this._openFrom);
+  }
+  return this;
+};
+
+
+/**
+ * Specifies where to animate the dialog close. `closeTo` accepts a
+ * query selector, DOM element, or a Rect object that is used to determine
+ * the bounds.
+ *
+ * @param {string|!Element|
+ *     {top: string, left: string, height: string, width: string}} closeTo
+ * @returns {MdPanelAnimation}
+ */
+MdPanelAnimation.prototype.closeTo = function(closeTo) {
+  this._closeTo = this._getBoundingClientRect(this._getElement(closeTo));
+  return this;
+};
+
+
+/**
+ * TODO(KarenParker): Switch and update the shared version of this method that
+ * others are currently building.
+ * Gets the element from the string|element|event.
+ * @param {string|!Element|
+ *     {top: string, left: string, height: string, width: string}} el
+ * @returns {!angular.JQLite}
+ * @private
+ */
+MdPanelAnimation.prototype._getElement = function(el) {
+  var queryResult = angular.isString(el) ?
+      document.querySelector(el) : el;
+  return angular.element(queryResult);
+};
+
+
+/**
+ * Identify the bounding RECT for the target element.
+ * @param {!angular.JQLite} element
+ * @returns {!Object}
+ * @private
+ */
+MdPanelAnimation.prototype._getBoundingClientRect = function(element) {
+  if (element && element.length) {
+    var bounds = { top:0, left:0, height:0, width:0 };
+    // "element" may just be a JQLite wrapper around the object with top/left/height/width.
+    var hasFn = angular.isFunction(element[0].getBoundingClientRect);
+
+    return {
+      element : hasFn ? element : undefined,
+      bounds  : hasFn ? element[0].getBoundingClientRect() : angular.extend({}, bounds, element[0]),
```

Comment:
super nit: I think this lines is too long. This file is using 80 characters, like the google style guide. (Same with the comment above and a few of your expects in the tests.)